gsd/__main__.py

```diff
@@ -0,0 +1,124 @@
+# Copyright (c) 2020 The Regents of the University of Michigan
+# All rights reserved.
+# This software is licensed under the BSD 3-Clause License.
+import sys
+import argparse
+import code
+
+from .version import __version__
+from .hoomd import open as hoomd_open, open_handle
+
+
+def _print_err(msg=None, *args):
+    print(msg, *args, file=sys.stderr)
+
+
+SHELL_BANNER = """Python {python_version}
+gsd {gsd_version}
+
+File: {fn}
+{additional_attributes}
+The GSD file handle is available via the "handle" variable.
+For supported schema, you may access the trajectory using the "traj" variable.
+Type "help(handle)" or "help(traj)" for more information.
+The gsd and gsd.fl packages are always loaded.
+Schema-specific modules (e.g. gsd.hoomd) are loaded if available."""
+
+
+def main_read(args):
+    # Default to a new line for well-formatted printing.
+    additional_attributes = "\n"
+
+    if args.schema == 'hoomd':
+        handle = open_handle(args.file, mode=args.mode)
```

Comment:
We don't need the schema if the mode is `ab` or `rb` (i.e. when we are not creating or overwriting a file). Should we try to use the fl module to open a handle from there, so we don't need to use only the `hoomd` schema? In `wb` or `xb` mode, though we do need the check since we cannot create a file without a schema.